app/models/concerns/foreman_azure_rm/vm_extensions/managed_vm.rb

```diff
@@ -13,18 +13,15 @@ def define_managed_storage_profile(vm_name, os_disk_caching, platform, premium_o
         os_disk.name = "#{vm_name}-osdisk"
         os_disk.os_type = platform
         os_disk.create_option = ComputeModels::DiskCreateOptionTypes::FromImage
-        os_disk.caching = if os_disk_caching.present?
-                            case os_disk_caching
-                              when 'None'
-                                ComputeModels::CachingTypes::None
-                              when 'ReadOnly'
-                                ComputeModels::CachingTypes::ReadOnly
-                              when 'ReadWrite'
-                                ComputeModels::CachingTypes::ReadWrite
-                            end
-                          else
-                            # ARM best practices stipulate RW caching on the OS disk
+        os_disk.caching = case os_disk_caching
```

Comment:
Can you please extract the case to a method, and use it twice - once here and once in `define_data_disks`

Reply:
done.